### PR TITLE
chore: add prepare query for delta/unspecified timeseries

### PR DESCRIPTION
**Pull request by srikanthccv**
### Summary

Part 3 of #4016 

Everything about the delta time series except `rate`/`increase` is the same as the cumulative. Since we already have the delta values we can simply sum them and divide them by the aggregation interval. If you recall the cumulative time series query the `lagInFrame` essentially calculates the delta first and then sums it up. Here we directly sum it up https://github.com/SigNoz/signoz/pull/4167/files#diff-382a6ec97dd3a6c236f66731e626db9b37b868e8a3d67c0f3bd09dd8ea43b604R69-R74

## Cumulative vs Delta

There are some cases where Cumulative shines, such as 

1. A failed export for delta is failed, but a failed export for Cumulative could be fixed by interpolation/extrapolation (not fully accurate as a real value, but a good approximation). 
2. Double entry of samples from retries wouldn't be an issue with Cumulative because we take the max of values from the time range, whereas the sum in delta might double count

However, unlike purpose-built systems like Prometheus, We don't exploit any of the characteristics or handle challenges that come with it. There is no reset handling, no interpolation/extrapolation, and more, mainly because ClickHouse doesn't have a way to handle them. We incline towards Delta because it's easier to handle; there is no need to worry about stale metrics because they are flushed for each interval; it is faster, less prone to memory issues, and more accurate because we don't need to deal with the complication that arises from resets etc...

### CODE REVIEW

**Review by nityanandagohain — DISMISSED** [review]

**Comment by srikanthccv** [comment]
@dhawal1248 please review

**Comment by srikanthccv** [comment]
Dhawal, Please review

**Comment by srikanthccv** [comment]
Screenshot 2023-12-21 at 2 23 05 PM

File: pkg/query-service/app/metrics/v4/delta/helper.go [hunk 1]
```diff
@@ -0,0 +1,57 @@
...
+	"strings"
+
+	v3 "go.signoz.io/signoz/pkg/query-service/model/v3"
+)
+
+// groupingSets returns a string of comma separated tags for group by clause
+// `ts` is always added to the group by clause
+func groupingSets(tags ...string) string {
+	withTs := append(tags, "ts")
+	return fmt.Sprintf(`GROUPING SETS ( (%s), (%s) )`, strings.Join(withTs, ", "), strings.Join(tags, ", "))
```
[the remark this answers is not in the document]
Comment (srikanthccv) [reply pkg/query-service/app/metrics/v4/delta/helper.go:14]: All queries whether it's cumulative or delta use GROUPING SETS

[the remark this answers is not in the document]
Comment (ankitnayan) [reply pkg/query-service/app/metrics/v4/delta/helper.go:14]: https://github.com/SigNoz/signoz/blob/develop/pkg/query-service/app/metrics/v4/cumulative/helper.go#L16-L18

[the remark this answers is not in the document]
Comment (srikanthccv) [reply pkg/query-service/app/metrics/v4/delta/helper.go:14]: When I initially worked on these stacked PRs they both were the same. We updated the ClickHouse version [recently](https://github.com/SigNoz/signoz/pull/4225) and it broke the GROUPING SETS functionality which we fixed in https://github.com/SigNoz/signoz/pull/4237. I thought I updated all places but this was left out as it did not get merged to develop. I will update it.

[the remark this answers is not in the document]
Comment (ankitnayan) [reply pkg/query-service/app/metrics/v4/delta/helper.go]: ok

**Review by ankitnayan — APPROVED** [review]

File: pkg/query-service/app/metrics/v4/delta/helper.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (srikanthccv) [reply pkg/query-service/app/metrics/v4/delta/helper.go]: Thanks for catching it.

**Pull request merged**